### PR TITLE
Add predicate to Unique() filter

### PR DESCRIPTION
**Pull request by atthom**
This should fix the issue #51

### CODE REVIEW

File: src/library.jl [hunk 1]
```diff
@@ -901,16 +901,26 @@ $(_thx_clj("distinct"))
...
  2
  3
 ```
 """
-struct Unique <: AbstractFilter
+struct Unique{P} <: AbstractFilter
+    pred::P
 end
 
+Unique() = Unique(x -> x)
```
Comment (tkf) [inline src/library.jl]:
```suggestion
Unique() = Unique(identity)
```

File: src/library.jl [hunk 2]
```diff
@@ -901,16 +901,26 @@ $(_thx_clj("distinct"))
 ```jldoctest
 julia> using Transducers
 
-julia> collect(Unique(), [1, 1, 2, 1, 3, 3, 2])
+julia> collect(Unique(), [1, 1, 2, -1, 3, 3, 2])
+3-element Array{Int64,1}:
```
Comment (tkf) [inline src/library.jl]:
```suggestion
4-element Array{Int64,1}:
```

This doctest is failing: https://travis-ci.com/tkf/Transducers.jl/jobs/240572228#L397

File: src/library.jl [hunk 3]
```diff
@@ -901,16 +901,26 @@ $(_thx_clj("distinct"))
...
+ 1
+ 2
+ -1
+ 3
+
+ julia> collect(Unique(x -> x^2), [1, 1, 2, -1, 3, 3, 2])
```
Comment (tkf) [inline src/library.jl]:
```suggestion
julia> collect(Unique(x -> x^2), [1, 1, 2, -1, 3, 3, 2])
```

File: src/library.jl [hunk 4]
```diff
@@ -901,16 +901,26 @@ $(_thx_clj("distinct"))
 ```jldoctest
 julia> using Transducers
 
-julia> collect(Unique(), [1, 1, 2, 1, 3, 3, 2])
+julia> collect(Unique(), [1, 1, 2, -1, 3, 3, 2])
+4-element Array{Int64,1}:
+ 1
```
Comment (tkf) [inline src/library.jl]:
```suggestion
  1
```

File: src/library.jl [hunk 5]
```diff
@@ -901,16 +901,26 @@ $(_thx_clj("distinct"))
...
+ 2
```
Comment (tkf) [inline src/library.jl]:
```suggestion
  2
```

File: src/library.jl [hunk 6]
```diff
@@ -901,16 +901,26 @@ $(_thx_clj("distinct"))
...
+ -1
+ 3
```
Comment (tkf) [inline src/library.jl]:
```suggestion
  3
```

**Comment by tkf** [comment]
(It's kind of annoying that GitHub does not have multi-line suggestion...)

By the way, can check `Allow edits from maintainers.` next time so that I can push the change?  I think you see something like this in the right sidebar:

> image

**Comment by atthom** [comment]
this seems like an endless process of checking for so few lines of codes!

**Comment by tkf** [comment]
I think it's good to go now.  Thanks for being patient :)

> It's already set ?

Oh, it was?  Then maybe I can't push it through GitHub web UI.  I guess I'll try using my git client next time.

**Pull request merged**